tests/test_openapi.py

```diff
@@ -45,3 +56,465 @@ def test_str2bool():
+
+
+def test_cql_filters_capabilities_assertion(get_cql_schemas, is_cql):
+    """
+    Assertions for CQL capabilities assertion schema in OpenAPI 3.0 Document
+
+    :param get_cql_schemas: OpenAPI 3.0 Document schemas
+    :param is_cql: boolean value
+    """
+    if is_cql:
+        # assertion for capabilities assertion
+        assert 'capabilities-assertion' in get_cql_schemas is not None
+        assert get_cql_schemas.get('capabilities-assertion', None) is not None
+
+        capabilities_assertion = ['name', 'operators', 'operands']
+        for props in capabilities_assertion:
+            assert props in get_cql_schemas['capabilities-assertion']['properties'] # noqa
+
+
+def test_cql_filters_function_description(get_cql_schemas, is_cql):
+    """
+    Assertions for CQL function description schema in OpenAPI 3.0 Document
+
+    :param get_cql_schemas: OpenAPI 3.0 Document schemas
+    :param is_cql: boolean value
+    """
+
+    if is_cql:
+        # assertion for function description
+        assert 'functionDescription' in get_cql_schemas is not None
+        assert get_cql_schemas.get('functionDescription', None) is not None
+
+        function_description = ['name', 'returns', 'arguments']
+        for props in function_description:
+            assert props in get_cql_schemas['functionDescription']['properties'] # noqa
+
+
+def test_cql_filters_capabilities(get_cql_schemas, is_cql):
+    """
+    Assertions for CQL filter capabilities schema in OpenAPI 3.0 Document
+
+    :param get_cql_schemas: OpenAPI 3.0 Document schemas
+    :param is_cql: boolean value
+    """
+    if is_cql:
+        # assertion for filter capabilities
+        assert 'filter-capabilities' in get_cql_schemas is not None
+        assert get_cql_schemas.get('filter-capabilities', None) is not None
+
+        filter_capabilities = ['conformance-classes',
+                               'capabilities', 'functions']
+        for props in filter_capabilities:
+            assert props in get_cql_schemas['filter-capabilities']['properties'] # noqa
+
+
+def test_auxiliary_openapi_extensions(get_cql_components, get_cql_schemas, is_cql):  # noqa
+    """
+    Assertions for auxiliary extensions in OpenAPI 3.0 Document
+
+    :param get_cql_components: OpenAPI 3.0 Document components
+    :param get_cql_schemas: OpenAPI 3.0 Document schemas
+    :param is_cql: boolean value
+    """
+    # assertion for components
+    assert get_cql_components is not None
+
+    # assertion for parameters
+    assert get_cql_components.get('parameters', None) is not None
+
+    # assertion for responses
+    assert get_cql_components.get('responses', None) is not None
+
+    # assertion for schemas
+    assert get_cql_schemas is not None
+
+    # assertion for queryables
+    assert get_cql_schemas.get('queryables', None) is not None
+    assert get_cql_components['responses'].get('Queryables', None) is not None
+
+    # assertion if filter is not supported
+    if not is_cql:
```

Comment:
In reference to https://github.com/geopython/pygeoapi/pull/478#pullrequestreview-443577941
@jorgejesus @francbartoli added test aux function to have assertion for non presence of filter